### PR TITLE
[KYUUBI #552] Sync build jobs between GA and travis

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Sync build jobs between GA and travis
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -30,20 +30,16 @@ matrix:
...
     - name: Test Kyuubi w/ -Pspark-3.0
       env:
-        - PROFILE="-Pspark-3.0"
-        - EXCLUDE_TAGS=""
+        - PROFILE="-Pspark-3.0 -Dspark.archive.mirror=https://archive.apache.org/dist/spark/spark-3.1.1 -Dspark.archive.name=spark-3.1.1-bin-hadoop2.7.tgz"
+        - EXCLUDE_TAGS="org.apache.kyuubi.tags.DataLakeTest"
     - name: Test Kyuubi w/ -Pspark-3.1
       env:
         - PROFILE="-Pspark-3.1"
-        - EXCLUDE_TAGS="org.apache.kyuubi.tags.DataLakeTest"
```
Comment (turboFei) [inline .travis.yml:38]: it is excluded by default, and keep it sync with GA

Comment (yaooqinn) [reply .travis.yml:38]: iceberg does not support spark 3.1 yet

Comment (turboFei) [reply .travis.yml:38]: I see. We need specify it here, otherwise, it will be overwritten by an empty string.

File: .travis.yml [hunk 2]
```diff
@@ -30,8 +30,8 @@ matrix:
   include:
```
Comment (pan3793) [inline .travis.yml:30]: GA also test against `PROFILE=''`

File: .travis.yml [hunk 3]
```diff
@@ -30,8 +30,8 @@ matrix:
...
     - name: Test Kyuubi w/ -Pspark-3.0
```
Comment (pan3793) [inline .travis.yml:31]: Test Kyuubi w/ -Pspark-3.0 and Spark 3.1 binary

File: .travis.yml (hunk 2 above)
Comment (turboFei) [reply .travis.yml:30]: so,  add tags into profile instead of EXECLUDE_TAGS?

Comment (yaooqinn) [reply .travis.yml:30]: PROFILE='' == PROFILE="-Pspark-3.0", this pr seems to delete the default job?

Comment (turboFei) [reply .travis.yml:30]: ok, got it

File: .travis.yml [hunk 4]
```diff
@@ -28,10 +28,14 @@ cache:
 
 matrix:
   include:
-    - name: Test Kyuubi w/ -Pspark-3.0
+    - name: Test Kyuubi w/ Default Profile
```
Comment (turboFei) [inline .travis.yml:31]: rename it to default profile to keep consistent with GA

**Review by pan3793 — APPROVED** [review]
LGTM

**Comment by turboFei** [comment]
The travis ci job was failed because its execution time was too long.

**Comment by turboFei** [comment]
merged to master, and will create a pr to split the build job.

**Pull request closed without merging**